Added files for easily running a headless backend.

SilvinWillemsen opened the pull request:
Input files are given in the headless_backend/input folder and results are written to the output folder

SilvinWillemsen committed:
Added files for easily running a headless backend. Input files are given in the headless_backend/input folder and results are written to the output folder

SilvinWillemsen committed:
Added main function to MyNewMethodInterface. Also changed IR length of DG to 0.01 for quicker prototyping.

mberz commented on simulation-backend/simulation_backend/headless_backend/HelperFunctions.py:
```diff
@@ -0,0 +1,66 @@
+    for subdir, dirs, files in os.walk(dirname):
+        for file in files:
+            # print os.path.join(subdir, file)
+            filepath = subdir + os.sep + file
```
```suggestion
            filepath = os.path.join(subdir, file)
```
How about this?

mberz commented on simulation-backend/simulation_backend/headless_backend/HelperFunctions.py:
```diff
@@ -0,0 +1,66 @@
+
+def find_input_file_in_subfolders(
+    dirname=os.path.dirname(__file__), file_name="MeasurementRoom.json"
+):
```
This function could use a docstring, which could be parsed by Sphinx.
See for example https://sphinxcontrib-napoleon.readthedocs.io/en/latest/example_numpy.html

mberz commented on simulation-backend/simulation_backend/headless_backend/HelperFunctions.py:
```diff
@@ -0,0 +1,66 @@
+                return filepath
+
+    print(f"{file_name} has NOT been found")
+    return ""
```
I'm not sure how these things are handled in the rest of the backend code.
In general I would think that explicitly raising an exception makes sense here if the file is not found. This way things cannot silently pass.
See https://realpython.com/python-exceptions/

```suggestion
    raise FileNotFoundError(f"{file_name} has NOT been found")
```

mberz commented on simulation-backend/simulation_backend/headless_backend/HelperFunctions.py:
```diff
@@ -0,0 +1,66 @@
+    return json_tmp_path
+
+
+def save_results(json_tmp_path):
```
I'm assuming `json_tmp_path` is a specific filename including the path, right? In that case it makes sense to rename the variable to reflect that.

SilvinWillemsen replied:
Hi @mberz, do you mean to rename it to json_tmp_file?

mberz commented on simulation-backend/simulation_backend/headless_backend/HelperFunctions.py:
```diff
@@ -0,0 +1,66 @@
+    def get_unique_filename(base_name="results", extension="json", subfolder="output"):
+        """Generate a unique filename inside a subfolder relative to this script."""
+        # Get the folder where the script is located
+        script_dir = os.path.dirname(os.path.abspath(__file__))
+
+        # Make sure subfolder exists
+        folder_path = os.path.join(script_dir, subfolder)
+        os.makedirs(folder_path, exist_ok=True)
+
+        # Start with results.json
+        filename = os.path.join(folder_path, f"{base_name}.{extension}")
+        counter = 1
+        while os.path.exists(filename):
+            filename = os.path.join(folder_path, f"{base_name} ({counter}).{extension}")
+            counter += 1
```
Not really important, but maybe easier to use: We could use the time the file is written as unique identifier instead of a number. That way it does not even need to scan the folder for existing files.

mberz reviewed (changes requested):
I'm only commenting on the changes made to the HelperFunctions.py.
It seems that the other files are not related to the PR.
I'm primarily missing docstrings (see comment on that) to get a better idea what a function is supposed to do and what input/output variables are. Please add these ;)

SilvinWillemsen committed:
Update simulation-backend/simulation_backend/headless_backend/HelperFunctions.py

Co-authored-by: Marco Berzborn <mberz@users.noreply.github.com>

SilvinWillemsen committed:
Update simulation-backend/simulation_backend/headless_backend/HelperFunctions.py

Co-authored-by: Marco Berzborn <mberz@users.noreply.github.com>

SilvinWillemsen committed:
Added Developer Guidelines to the backend repository.
Changed the uuids in the .geo, .msh and .json files to human-readable names
Added impulse response to DG results
Added gmsh.is_initialised() check when initialising in FVMinterface (DE interface)
Renamed .json files to exampleInput_<method_acronym>

SilvinWillemsen committed:
Merge branch 'headlessbackend' of github.com:choras-org/backend into headlessbackend

SilvinWillemsen committed:
Renamed load_tmp_from_input to create_tmp_from_input
Added docstrings to HelperFunctions.py
